### PR TITLE
New `completions <shell>` subcommand

### LINKED ISSUE #1947: New Subcommand for Generating Shell Completions
I'm proposing a new subcommand that can generate shell completions. The `clap` crate is already being used for actually building the `mirrord` CLI, and they already build a crate [`clap_complete`](https://github.com/clap-rs/clap/tree/master/clap_complete) to make generating shell completions simple for a few popular shells (bash, fish, zsh, powershell, elvish). This would make discoverability of flags and subcommands much simpler to newcomers and as the number of options grow.

It looks like some subcommands are hidden (like `extract` and `verify`) - I don't know if clap_complete supports _not_ generating completions for them, or if that would be an issue if they were.

It would look like:

```bash
$ mirrord completions bash
### long output of bash completion to stdout truncated here

## could directly source
$ source <(mirrord completions bash)

$ mirrord exec <TAB><TAB>
-t                             -c                             --agent-namespace              --skip-processes               --no-outgoing                  --help
-n                             -e                             --agent-log-level              --agent-ttl                    --no-tcp-outgoing              <BINARY>
-a                             -p                             --agent-image                  --agent-startup-timeout        --no-udp-outgoing              [BINARY_ARGS]...
-l                             -f                             --fs-mode                      --accept-invalid-certificates  --no-telemetry                 
-i                             -h                             --override-env-vars-exclude    --ephemeral-container          --disable-version-check        
-x                             --target                       --override-env-vars-include    --steal                        --config-file                  
-s                             --target-namespace             --no-remote-dns                --pause                        --context   

$ mirrord exec --fs-mode <TAB><TAB>
write                 read                  local                 local-with-overrides  
````

I know Homebrew has some nice functionality for calling these scripts and putting the generated output/files to sane places to source from when mirrord is installed, not sure about other package managers.

Happy to do the implementation myself, it should be fairly straightforward.

### PR DESCRIPTION
**Pull request by bradschwartz**
Closes #1947

Adding a new subcommand for generating shell completion scripts. It can be invoked like: `mirrord completions <bash|fish|powershell|zsh|elvish>` This will make discoverability of subcommands and flags simpler for newcomers. I know Homebrew has some built-in support for calling these types of methods and installing completion scripts in the relevant places, and I see it done enough like: `source <(mirrord completions bash)` in .bashrc scripts

Everything is generated via the clap_complete crate, so should stay up to date with new options.

Currently the hidden commands are coming up in shell completions:

```bash
$ mirrord ex <TAB><TAB>
exec     extract  ext
```

Thank you for contributing to mirrord!

Please make sure you added a CHANGELOG file in `changelog.d/` named `issue_number.category.md`.
For example, `1054.changed.md` or `+towncrier.added.md` (if no issue).

### CODE REVIEW

File: changelog.d/1947.fixed.md [hunk 1]
```diff
@@ -0,0 +1 @@
+New subcommand for generating shell completions for bash/fish/zsh/powershell/elvish
```
Comment (aviramha) [inline changelog.d/1947.fixed.md:1]: Please change to `1947.added.md`

**Review by aviramha — CHANGES_REQUESTED** [review]
Thanks for this! I've left one comment regarding the changelog file name.

File: changelog.d/1947.fixed.md (hunk 1 above)
Comment (bradschwartz) [reply changelog.d/1947.fixed.md]: Just renamed 👍

**Review by aviramha — APPROVED** [review]

**Comment by aviramha** [comment]
Great, Thank you!

**Pull request merged**